src/hotspot/share/nmt/nmtTreap.hpp

```diff
@@ -209,12 +210,13 @@ class Treap {
       seen_count++;
       if (last_seen == nullptr) {
         last_seen = node;
-        return;
+        return true;
       }
       if (COMPARATOR::cmp(last_seen->key(), node->key()) > 0) {
         failed = false;
       }
       last_seen = node;
+      return true;
```

Comment:
Why are we returning a `bool` value in `void` function?

Reply:
That is returning from the lambda function.
In `visit_in_order(F func)`, when function `func` (in our case the lambda) returns `false` it means the iteration should stop. Otherwise, if it returns `true` it means to continue iterating.